sycl/include/sycl/handler.hpp

```diff
@@ -488,6 +488,9 @@ class __SYCL_EXPORT handler {
   /// \param Graph is a SYCL command_graph
   handler(std::shared_ptr<ext::oneapi::experimental::detail::graph_impl> Graph);
 #endif
+  handler(std::unique_ptr<detail::handler_impl> &&HandlerImpl);
+
+  ~handler();
```

Comment:
`std::unique_ptr<T>` dtor needs a complete `T`, so move the definition to the `handler.cpp`.